src/validation.cpp

```diff
@@ -3649,7 +3662,28 @@ bool BlockManager::LoadBlockIndex(
         if (pindex->IsAssumedValid() ||
                 (pindex->IsValid(BLOCK_VALID_TRANSACTIONS) &&
                  (pindex->HaveTxsDownloaded() || pindex->pprev == nullptr))) {
-            block_index_candidates.insert(pindex);
+
+            // Fill each chainstate's block candidate set. Add all blocks as
+            // candidates if the chainstate is allowed to rely on assumed-valid
+            // blocks. Otherwise avoid adding assumed-valid blocks so the
+            // chainstate will download and validate all earlier blocks.
```

Comment:
I think the comment can be slightly modified to mention the descendants of assumed-valid blocks, like in the commit description.

"Otherwise avoid adding assumed-valid blocks and their fully-validated descendants therefore inducing the chainstate to download and validate all blocks earlier than the first assumed-valid block".

Reply:
Not sure I follow your suggested comment, since fully-validated blocks should be added to both chainstates, but I've updated the comment to be try and be clearer.